### PR TITLE
Dynamically and use filtered and trimmed list of regions

### PR DESCRIPTION
**Pull request by aimonb**
- Aimon Bustardo

Some small fixes to the network creation

        - Jeffrey Dunham <jeffrey.a.dunham@gmail.com>

Quick doc fix
        - Jeffrey Dunham <jeffrey.a.dunham@gmail.com>

Change-Id: I70db3fa0bb161170ab7d73a0b9f2f02535dbbacb

### CODE REVIEW

File: ansible/dimensiondata/dimensiondata_network.py [hunk 1]
```diff
@@ -41,7 +44,7 @@
   region:
     description:
       - The target region.
-    choices: ['na', 'eu', 'au', 'af', 'ap', 'latam', 'canada', 'canberra', 'id', 'in', 'il', 'sa']
+    choices: %s
```
Comment (jadunham1) [inline ansible/dimensiondata/dimensiondata_network.py]: I don't see the actual string coming in.

Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_network.py]: Ah crap.. bad editing on rebase conflict. re-adding missing part.

File: ansible/module_utils/dimensiondata.py [hunk 2]
```diff
@@ -65,3 +67,22 @@ def get_credentials():
...
+def get_dd_regions():
+    # Get endpoints
+    all_regions = API_ENDPOINTS.keys()
+
+    # filter to only DimensionData endpoints
+    filt = re.compile(r'^dd-')
+    region_list = filter(lambda i: filt.search(i), all_regions)
+
+    # Strip prefix
+    regions = [re.sub(r'^[a-z]+([0-9]+)?-', '', region)
```
Comment (jadunham1) [inline ansible/module_utils/dimensiondata.py]: Over regex's in python I would use string methods.

I.e.
if region.startswith('dd-'):
   region = region[:3]

Comment (jadunham1) [inline ansible/module_utils/dimensiondata.py]: Sorry region[3:]

File: ansible/dimensiondata/dimensiondata_network.py [hunk 3]
```diff
@@ -72,7 +75,7 @@
       - Should the resource be present or absent.
     choices: [present, absent]
     default: present
-'''
+''' % str(dd_regions)
```
Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_network.py]: Here is the missing bit from last commit.

File: ansible/module_utils/dimensiondata.py (hunk 2 above)
Comment (aimonb) [inline ansible/module_utils/dimensiondata.py]: Ah good point, I forget about that method. will adjust.

**Pull request closed without merging**